src/router.tsx

```diff
@@ -14,20 +14,13 @@ interface Route {
   component: () => Promise<ComponentModule>
 }
 
-interface URLComponentPromiseMap {
-  [url: string]: () => Promise<ComponentModule>
-}
-
-interface URLComponentMap {
-  [url: string]: AnyComponent
-}
-
 const routers: ((url: string) => void)[] = []
 
 export const route = (url: string) => {
   routers.forEach(router => {
     router(url)
   })
+  history.pushState(null, '', url)
```

Comment:
I refactored this component because I was pretty sure the bug was in here. It turned out the bug was not in here, but my refactorings made it smaller, more correct, and simpler, so I kept them